app/src/androidTest/java/com/github/houseorganizer/houseorganizer/EditHouseholdTest.java

```diff
@@ -0,0 +1,389 @@
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.intent.Intents.intended;
+import static androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent;
+import static androidx.test.espresso.matcher.ViewMatchers.isClickable;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static com.github.houseorganizer.houseorganizer.FirebaseTestsHelper.TEST_HOUSEHOLD_NAMES;
+import static com.github.houseorganizer.houseorganizer.FirebaseTestsHelper.TEST_USERS_EMAILS;
+import static com.github.houseorganizer.houseorganizer.FirebaseTestsHelper.UNKNOWN_USER;
+import static com.github.houseorganizer.houseorganizer.FirebaseTestsHelper.WRONG_EMAIL;
+import com.github.houseorganizer.houseorganizer.house.HouseSelectionActivity;
+import com.github.houseorganizer.houseorganizer.house.EditHousehold;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import android.content.Intent;
+
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.espresso.intent.Intents;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+@RunWith(AndroidJUnit4.class)
+public class EditHouseholdTest {
+    private static FirebaseFirestore db;
+    private static FirebaseAuth auth;
+    private static Intent intentFromHouseSelection;
+
+    @BeforeClass
+    public static void createMockFirebase() throws ExecutionException, InterruptedException {
+        FirebaseTestsHelper.startAuthEmulator();
+        FirebaseTestsHelper.startFirestoreEmulator();
+        FirebaseTestsHelper.setUpFirebase();
+
+        db = FirebaseFirestore.getInstance();
+        auth = FirebaseAuth.getInstance();
+        intentFromHouseSelection = new Intent(ApplicationProvider.getApplicationContext(), EditHousehold.class).putExtra(HouseSelectionActivity.HOUSEHOLD_TO_EDIT, TEST_HOUSEHOLD_NAMES[0]);
+    }
+
+    @AfterClass
+    public static void signOut(){
+        auth.signOut();
+    }
+
+    @Rule
+    public ActivityScenarioRule<EditHousehold> editHouseholdRule = new ActivityScenarioRule<>(intentFromHouseSelection);
+
+    @Before
+    public void setupHouseholds() throws ExecutionException, InterruptedException {
+        FirebaseTestsHelper.createHouseholds();
+    }
+
+    @After
+    public void dismantleHouseholds() throws ExecutionException, InterruptedException {
+        FirebaseTestsHelper.createHouseholds();
```

Comment:
Just caught this one - shouldn't you delete the data instead of creating it again?

Reply:
I think the function just has a confusing name because actually it sets up the households to their original state.
It creates them only if they don't exist, otherwise it resets them

Reply:
Yes @Geno22 is right. The create household does not care about the state of the data base as the state is different after each test. It does not try to create, but rathers sets the data to X without caring about any state whatsoever

Reply:
OK, I see now. The thing is, you can add this to @AfterClass instead. Since the @Before already calls the same method for each test, an @After seems like a surplus to me in this context?

Reply:
On it